app/cli/cmd/root.go

```diff
@@ -239,46 +268,125 @@ func loadControlplaneAuthToken(cmd *cobra.Command) (string, error) {
 // 3. API token
 // Each one of them have an associated audience claim that we use to identify the type of token. If the token is not
 // present, nor we cannot match it with one of the expected audience, return nil.
-func parseToken(token string) (*ParsedToken, error) {
+func parseToken(token string) (*parsedToken, error) {
 	if token == "" {
-		return &ParsedToken{}, nil
+		return nil, nil
 	}
 
 	// Create a parser without claims validation
 	parser := jwt.NewParser(jwt.WithoutClaimsValidation())
 
 	// Parse the token without verification
-	parsedToken, _, err := parser.ParseUnverified(token, jwt.MapClaims{})
+	t, _, err := parser.ParseUnverified(token, jwt.MapClaims{})
 	if err != nil {
 		return nil, err
 	}
 
-	// Extract claims
-	claims := parsedToken.Claims.(jwt.MapClaims)
+	// Extract generic claims otherwise, we would have to parse
+	// the token again to get the claims for each type
+	claims, ok := t.Claims.(jwt.MapClaims)
+	if !ok {
+		return nil, nil
+	}
 
 	// Get the audience claim
-	if val, ok := claims["aud"]; ok && val != nil {
-		// Chainloop tokens have only one audience in an array
-		aud, ok := val.([]interface{})
-		if !ok {
-			return nil, nil
+	val, ok := claims["aud"]
+	if !ok || val == nil {
+		return nil, nil
+	}
+
+	// Ensure audience is an array of interfaces
+	// Chainloop only has one audience per token
+	aud, ok := val.([]interface{})
+	if !ok || len(aud) == 0 {
+		return nil, nil
+	}
+
+	// Initialize parsedToken
+	pToken := &parsedToken{}
+
+	// Determine the type of token based on the audience.
+	switch aud[0].(string) {
+	case apiTokenAudience:
+		pToken.Type = "api-token"
+		if tokenID, ok := claims["jti"].(string); ok {
+			pToken.ID = tokenID
 		}
-		if len(aud) == 0 {
-			return nil, nil
+		if orgID, ok := claims["org_id"].(string); ok {
+			pToken.OrgID = orgID
 		}
-
-		switch aud[0].(string) {
-		case apiTokenAudience:
-			return &ParsedToken{Type: "api-token"}, nil
-		case userAudience:
-			userID := claims["user_id"].(string)
-			return &ParsedToken{Type: "user", ID: userID}, nil
-		case robotAccountAudience:
-			return &ParsedToken{Type: "robot-account"}, nil
-		default:
-			return nil, nil
+	case userAudience:
+		pToken.Type = "user"
+		if userID, ok := claims["user_id"].(string); ok {
+			pToken.ID = userID
+		}
+	case robotAccountAudience:
+		pToken.Type = "robot-account"
+		if tokenID, ok := claims["jti"].(string); ok {
+			pToken.ID = tokenID
+		}
+		if orgID, ok := claims["org_id"].(string); ok {
+			pToken.OrgID = orgID
 		}
+	default:
+		return nil, nil
+	}
+
+	return pToken, nil
+}
+
+var (
```

Comment:
Not important: Do you think we can move the below code to its own file `telemetry.go` or similar just for readability?

Reply:
I would do it if we really support more telemetry and figure out the problem with the logger since right now it's coupled to it.